src/Features/CSharp/Portable/ValidateFormatString/CSharpValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpValidateFormatStringDiagnosticAnalyzer : 
+        AbstractValidateFormatStringDiagnosticAnalyzer<SyntaxKind>
+    {
+        protected override ISyntaxFactsService GetSyntaxFactsService()
+            => CSharpSyntaxFactsService.Instance;
+
+        internal override bool TryGetArgsArgumentType(
+            SemanticModel semanticModel,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters,
+            out ITypeSymbol argsArgumentType)
+        {
+            if (!TryGetArgument(semanticModel, "args", arguments, parameters, out var argsArgument))
+            {
+                argsArgumentType = null;
+                return false;
+            }
+
+            argsArgumentType = semanticModel.GetTypeInfo(argsArgument.Expression).Type;
+            return true;
+        }
+
+        private static bool TryGetArgument(
+            SemanticModel semanticModel,
+            string searchArgumentName,
+            SeparatedSyntaxList<ArgumentSyntax> arguments,
+            ImmutableArray<IParameterSymbol> parameters,
+            out ArgumentSyntax argumentSyntax)
+        {
+            argumentSyntax = null;
+
+            // First, look for a named argument that matches
+            var namedArgument = arguments.SingleOrDefault(
+                p => p.NameColon?.Name.Identifier.ValueText.Equals(searchArgumentName) == true);
+
+            if (namedArgument != null)
+            {
+                argumentSyntax = namedArgument;
+                return true;
+            }
+
+            // If no named argument exists, look for the named parameter
+            // and return the corresponding argument
+            var namedParameter = parameters.SingleOrDefault(p => p.Name.Equals(searchArgumentName) == true);
```

Comment:
== true   is unnecessary.
SingleOrDefault may throw if tehre are multiple parameters with teh same name.